auth/standard_role_manager.cc

```diff
@@ -701,4 +714,137 @@ future<> standard_role_manager::remove_attribute(std::string_view role_name, std
                 {sstring(role_name), sstring(attribute_name)});
     }
 }
+
+future<std::vector<role_info>> standard_role_manager::get_role_info(const bool with_salted_hashes) const {
+    // This is super verbose, but it's difficult to avoid it.
+    // Unfortunately, fmt isn't able to produce `constexpr std::string_view`s.
+    const auto role_query = std::invoke([&] {
+        if (with_salted_hashes) {
+            return seastar::format("SELECT {}, {}, {}, {} FROM {}.{}",
+                meta::roles_table::role_col_name,
+                meta::roles_table::can_login_col_name,
+                meta::roles_table::is_superuser_col_name,
+                meta::roles_table::salted_hash_col_name,
+                get_auth_ks_name(_qp),
+                meta::roles_table::name);
+        }
+
+        return seastar::format("SELECT {}, {}, {} FROM {}.{}",
```

Comment:
why you decided to write separate selects and not reusing existing auth code?

Reply:
Which piece of code do you mean? Unfortunately, I didn't find any existing functions responsible for querying all of this information in one go, so I went with this approach

Reply:
yes, sure there will be no such singe place, but in theory all this info should be accessible via existing API (otherwise it would be unused)

Reply:
@nuivall I think the biggest benefit is latency/efficiency - it's cheaper to do a single query rather then e.g. a separate query for each item. In enterprise, we also have an LDAP role manager which sends queries to an external service, and from what I've heard it is possible to implement some of these operations as a single query.

Reply:
I'd say code reuse overtake efficiency in this particular case.